packages/ui-react/src/components/Modal/Modal.tsx

```diff
@@ -49,9 +49,10 @@ const Modal: React.FC<Props> = ({ open, onClose, children, AnimationComponent =
       document.body.style.marginRight = `${scrollbarSize()}px`;
       document.body.style.overflowY = 'hidden';
 
-      // focus the first element in the modal
+      // focus the first element in the modal (except the close button)
       if (modalRef.current) {
-        const interactiveElement = modalRef.current.querySelectorAll('a, button, [tabindex="0"]')[0] as HTMLElement | null;
+        const interactiveElements = modalRef.current.querySelectorAll('a, button, [tabindex="0"]');
+        const interactiveElement = Array.from(interactiveElements).filter((item) => !item.hasAttribute('data-return-button'))[0] as HTMLElement | null;
```

Comment:
Minor: Pure CSS selector?

```suggestion
        const interactiveElement = modalRef.current.querySelectorAll('a:not([data-close), button:not([data-close), [tabindex="0"]:not([data-close)')[0];
```

Reply:
Chose an alternative approach